src/android/PurchasesPlugin.java

```diff
@@ -156,6 +156,11 @@ private void setAutomaticAppleSearchAdsAttributionCollection(boolean enabled, Ca
         // NOOP
     }
 
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setupShouldPurchasePromoProductCallback")
+    private void setupShouldPurchasePromoProductCallback(CallbackContext callbackContext) {
+        // NOOP
```

Comment:
wild violation of Interface Segregation Principle, but I think it's okay? I copied it from line 156. 
other ideas would be to have the javascript side decide (which isn't ideal), or to somehow split interfaces, which might be too much work for too little gain.

Reply:
I think this is better too